tests/unit/commands/local/lib/test_cfn_api_provider.py

```diff
@@ -0,0 +1,215 @@
+            Api(path="/path1", method="POST", function_name="SamFunc1", cors=None),
+
+            Api(path="/path2", method="PUT", function_name="SamFunc1", cors=None),
+            Api(path="/path2", method="GET", function_name="SamFunc1", cors=None),
+
+            Api(path="/path3", method="DELETE", function_name="SamFunc1", cors=None)
+        ]
+
+    def test_with_no_apis(self):
+        template = {
+            "Resources": {
+
+                "Api1": {
+                    "Type": "AWS::ApiGateway::RestApi",
+                    "Properties": {
+                    },
+
+                }
+            }
+        }
+
+        provider = ApiProvider(template)
+
+        self.assertEquals(provider.apis, [])
+
+    def test_with_inline_swagger_apis(self):
+        template = {
+            "Resources": {
+
+                "Api1": {
+                    "Type": "AWS::ApiGateway::RestApi",
+                    "Properties": {
+                        "Body": make_swagger(self.input_apis)
+                    }
+                }
+            }
+        }
+
+        provider = ApiProvider(template)
+        assertCountEqual(self, self.input_apis, provider.apis)
+
+    def test_with_swagger_as_local_file(self):
+        with tempfile.NamedTemporaryFile(mode='w') as fp:
+            filename = fp.name
+
+            swagger = make_swagger(self.input_apis)
+            json.dump(swagger, fp)
+            fp.flush()
+
+            template = {
+                "Resources": {
+
+                    "Api1": {
+                        "Type": "AWS::ApiGateway::RestApi",
+                        "Properties": {
+                            "BodyS3Location": filename
+                        }
+                    }
+                }
+            }
+
+            provider = ApiProvider(template)
+            assertCountEqual(self, self.input_apis, provider.apis)
+
+    def test_body_with_swagger_as_local_file_expect_fail(self):
+        with tempfile.NamedTemporaryFile(mode='w') as fp:
+            filename = fp.name
+
+            swagger = make_swagger(self.input_apis)
+            json.dump(swagger, fp)
+            fp.flush()
+
+            template = {
+                "Resources": {
+
+                    "Api1": {
+                        "Type": "AWS::ApiGateway::RestApi",
+                        "Properties": {
+                            "Body": filename
+                        }
+                    }
+                }
+            }
+            self.assertRaises(Exception, ApiProvider, template)
+
+    @patch("samcli.commands.local.lib.cfn_base_api_provider.SamSwaggerReader")
+    def test_with_swagger_as_both_body_and_uri(self, SamSwaggerReaderMock):
+        body = {"some": "body"}
+        filename = "somefile.txt"
+
+        template = {
+            "Resources": {
+
+                "Api1": {
+                    "Type": "AWS::ApiGateway::RestApi",
```

Comment:
Is this even valid? Would deploying a template like this fail in CloudFormation?

Reply:
Its missing the StageName property, but seems to be valid other than that. I'll add that.